xblock/mixins.py

```diff
@@ -522,7 +522,9 @@ def _add_field(self, node, field_name, field):
             elem.text = value
             node.insert(0, elem)
         else:
-            node.set(field_name, value)
+            # setting the None in the value for node raises TypeError to avoid this
```

Comment:
Just to be clear, that seems like the bug to me. Why is `to_string` not returning a string?